lib/utils/resolveSilent.mjs

```diff
@@ -0,0 +1,46 @@
+import fs from 'node:fs';
+import path from 'node:path';
+
+import { fileURLToPath, pathToFileURL } from 'node:url';
+
+import { moduleResolve } from '@dual-bundle/import-meta-resolve';
+import resolveFrom from 'resolve-from';
+
+const pathSuffixes = ['', '.js', '.json', `${path.sep}index.js`, `${path.sep}index.json`];
+
+const specifierSuffixes = ['', '.js', '.json', '/index.js', '/index.json'];
+
+/**
+ * @param {string} parent
+ * @param {string} lookup
+ * @return {string | undefined}
+ */
+export default function resolveSilent(parent, lookup) {
+	if (path.isAbsolute(lookup)) {
+		for (const suffix of pathSuffixes) {
+			const filename = lookup + suffix;
```

Comment:
[ask] I don't understand the necessity of the suffixes. Can you explain?

Reply:
In pure ESM, those anonymous suffixes will not be tried automatically anymore while for commonjs (`resolve-from`) will try them automatically, so for compatibility we need to try all theses files inside.

Reply:
> I doubt will we remove the compatibility in the future

Why? In pure ESM, supporting these implicit suffixes (e.g. `index.js`) will not make sense, right? Because `import` requires an accurate path unlike `require()`?

Reply:
That's good if we don't need such compatibility. I'll add such comment.

```
TODO: These suffixes are here for commonjs compatibility reason, we should remove these codes after migrating to pure ESM because in ESM accurate paths are required
```

Reply:
Sounds good 👍🏼